skyportal/tests/conftest.py

```diff
@@ -145,54 +166,79 @@ def take_screenshot_and_page_source(webdriver, nodeid):
 
 @pytest.fixture()
 def public_stream():
-    return StreamFactory()
+    stream = StreamFactory()
+    # Save ID of new DB row
+    stream_id = stream.id
+    yield stream
+    StreamFactory.teardown(stream_id)
 
 
 @pytest.fixture()
 def public_stream2():
-    return StreamFactory()
+    stream = StreamFactory()
+    stream_id = stream.id
+    yield stream
+    StreamFactory.teardown(stream_id)
 
 
 @pytest.fixture()
 def stream_with_users(super_admin_user, group_admin_user, user, view_only_user):
-    return StreamFactory(
+    stream = StreamFactory(
         users=[super_admin_user, group_admin_user, user, view_only_user]
     )
+    stream_id = stream.id
+    yield stream
+    StreamFactory.teardown(stream_id)
 
 
 @pytest.fixture()
 def public_group(public_stream):
-    return GroupFactory(streams=[public_stream])
+    group = GroupFactory(streams=[public_stream])
+    group_id = group.id
+    yield group
+    GroupFactory.teardown(group_id)
 
 
 @pytest.fixture()
 def public_group2(public_stream):
-    return GroupFactory(streams=[public_stream])
+    group = GroupFactory(streams=[public_stream])
+    group_id = group.id
+    yield group
+    GroupFactory.teardown(group_id)
 
 
 @pytest.fixture()
 def public_group_no_streams():
-    return GroupFactory()
+    group = GroupFactory()
+    group_id = group.id
+    yield group
+    GroupFactory.teardown(group_id)
 
 
 @pytest.fixture()
 def group_with_stream(
     super_admin_user, group_admin_user, user, view_only_user, public_stream
 ):
-    return GroupFactory(
+    group = GroupFactory(
         users=[super_admin_user, group_admin_user, user, view_only_user],
         streams=[public_stream],
     )
+    group_id = group.id
+    yield group
+    GroupFactory.teardown(group_id)
```

Comment:
The pattern everywhere seems to be identical, so it might be possible to make this a generic fixture, something like:

```
def DBObjectFixture(obj_factory):
    obj = yield from obj_factory
    yield obj
    obj.__class__.teardown(obj.id)
```

And then:

```
@pytest.fixture
@ObjFixture
def group_with_stream(...):
    group = ...
    return group
```